### PR TITLE
DDCE-2733 - Migrate from reactive mongo to hmrc mongo

### PR DESCRIPTION
**Pull request by frikit**

### CODE REVIEW

**Review by ina-tu — COMMENTED** [review]
The back buttons on the pages have disappeared - see screenshots below:
Left is local environment, Right is Staging environment
Screenshot 2022-11-02 at 17 02 18

Screenshot 2022-11-02 at 17 02 33

**Review by ina-tu — CHANGES_REQUESTED** [review]
when the link in the error summary is clicked, the "#value yes" (hyperlink) appears in the url (http://localhost:8831/verify-your-identity-for-an-estate/managed-by-agent#value-yes). This doesn't happen in staging environment - see screenshot below:

Left is local, Right is staging

Screenshot 2022-11-02 at 17 11 51

File: app/repositories/DefaultSessionRepository.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        IndexOptions()
+          .unique(false)
+          .name("user-answers-updated-at-index")
+          .expireAfter(config.cacheTtlSeconds, TimeUnit.SECONDS))),
+    replaceIndexes = config.dropIndexes
+  )
+    with SessionRepository {
+
+  override def get(id: String): Future[Option[UserAnswers]] =
+    collection.find(Filters.eq("_id", id)) headOption()
```
[the remark this answers is not in the document]
Comment (frikit) [reply app/repositories/DefaultSessionRepository.scala:52]: I fixed as I think is a good change it but in scala it is a valid syntax does the same as your example

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -1,40 +1,30 @@
...
-    "org.jsoup"                   %  "jsoup"                    % "1.15.1",
-    "com.typesafe.play"           %% "play-test"                % PlayVersion.current,
-    "org.mockito"                 %% "mockito-scala"            % "1.17.12",
-    "org.scalacheck"              %% "scalacheck"               % "1.16.0",
-    "com.github.tomakehurst"      % "wiremock-standalone"       % "2.27.2",
-    "com.vladsch.flexmark"        % "flexmark-all"              % "0.62.2"
+    "org.jsoup"                   %  "jsoup"                    % "1.15.3",
+    "org.scalatest"               %% "scalatest"                % "3.2.14",
+    "org.mockito"                 %% "mockito-scala-scalatest"  % "1.17.12",
+    "org.scalatestplus.play"      %% "scalatestplus-play"       % "5.1.0",
```
[the remark this answers is not in the document]
Comment (frikit) [reply project/AppDependencies.scala:23]: done

**Comment by frikit** [comment]
@ina-tu fixed comments and reverted play-frontend-hmrc to avoid ui bugs

**Review by ina-tu — APPROVED** [review]

**Pull request merged**